src/components/error_boundary/errors.ts

```
@@ -0,0 +1,43 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { $Values } from 'utility-types';
+
+export const ErrorType = Object.freeze({
+  Graceful: 'graceful' as const,
+});
+/** @public */
+export type ChartTypes = $Values<typeof ErrorType>;
+
+/**
+ * Error to used to gracefully render empty chart
+ */
+export class GracefulError extends Error {
+  type = ErrorType.Graceful;
+  renderEmptyChart: boolean;
```

This flag is not currently used, what was the idea behind that?

---

Thanks, I'll remove it for now. The idea was to allow this to be a more generic error. You could throw this error everywhere is charts and customize how the error is handled such as if it should throw, or if the chart should render empty. I think this can be discussed once we improve the error handling.